### DISCUSSION TITLE
Upcoming changes to Conan Center CI workflow

### DISCUSSION THREAD

**Discussion by jcar87** [conan-io/conan-center-index#25772] [Announcements]
Coinciding with the deprecation of Conan 1.x support in Conan Center (see [announcement](https://github.com/conan-io/conan-center-index/discussions/25461)) - we will begin rolling out updates to the CI workflow in the conan-center-index Github repository. Our goal is to provide a much improved experience for pull request contributors and reviewers.

## Upcoming changes (week starting 4 November 2024):

* Workflow for new contributors
  - New contributors are no longer required to request authorization in a separate PR. Conan team maintainers can approve a CI run based on the changes in the diff.
* Improved CI build times
  - bottlenecks have been identified and mitigated in order to provide a quicker turnaround time.
  - Better use of available compute resources.
* Improved reliability
 - The causes of many sporadic CI failures have been identified and addressed.
* CI runs are no longer restarted when the PR description is modified
  - Unified and improved results reporting
  - All results are now reported in the Checks tab, rather than PR comments. This includes: build results, linter warnings, hook results
  - Build results are reported immediately as they become available (per package built) - the page is updated on the fly as new packages are built, rather than at the end.
* Redesigned Conan recipe linter
  - The linter rules have been tailored to detect common recipe mistakes. The primary goal is bug prevention.
  - The linter covers the recipe, as well as config.yml and conandata.yml
* Improved pre-validation and error reporting
  - Additional checks are performed before building any package - the job is stopped if there are any critical issues
  - Clearer summary of which packages are expected to be built - and when can’t or they are not expected to be built, details are provided, including the reason and which configurations are impacted.
* Improved reporting from hook warnings
  - Relevant Conan hooks have been ported to Conan 2
  - Hook warnings are now clearly reported per package built, as well as a summary at the end of the build
* ❗If a PR targets files that have already been modified in the head of the master branch, it will not be mergeable and will require the PR branch to be synchronized with the master branch and go through CI checks again - even when there are no merge conflicts

### Ongoing work to be rolled out in coming weeks and months:
 - Unified queue: when the system is under maintenance and not accepting jobs, pending jobs are in a queue and once the service resumes will be
- Support up-to-date versions of compilers

> [!NOTE]  
> ### Quick summary of changes for contributors:
> - Starting 4-November-2024 Conan Center CI will no longer support Conan 1.x
>     - We emphatically ask to not perform refactorings or cleanup in recipes as we continue working through the backlog
>   - CI results are no longer reported as PR comments - all results are published in the “Checks” tab
>   - If a PR targets files that have already been modified in the head of the master branch, it will not be mergeable and will require the PR branch to be synchronized with the master branch - even when there are no merge conflicts

**Comment by ericLemanissier** [comment]
previous CI used to test the recipe on all supported platforms, cf https://c3i.jfrog.io/c3i/misc-v2/summary.html?json=https://c3i.jfrog.io/c3i/misc-v2/logs/pr/25812/1/msgpack-cxx/7.0.0//summary.json even if there was a single Package ID.
It looks like the new CI does not run more test than valid package ID  in https://github.com/conan-io/conan-center-index/pull/25812/checks
Does it mean that header only recipes will never be evaluated in Windows and MacOs context ?

<sub>reposted from https://github.com/conan-io/conan-center-index/pull/25812#issuecomment-2454397961</sub>
